timesketch/frontend-ng/src/views/Graph.vue

```diff
@@ -0,0 +1,658 @@
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+-->
+<template>
+  <v-container fluid>
+    <v-card flat class="pa-3 pt-0 mt-n3" color="transparent">
+      <v-card class="d-flex align-start mb-1" outlined>
+        <v-text-field
+          v-model="filterString"
+          @input="filterGraphByInput"
+          class="pa-1"
+          placeholder="Filter nodes and edges"
+          label="Filter nodes and edges"
+          append-icon="mdi-magnify"
+          hide-details
+          single-line
+          dense
+          filled
+          flat
+          solo
+        >
+        </v-text-field>
+      </v-card>
+      <v-toolbar dense flat color="transparent">
+        <div>
+          <span class="mr-2">
+            <small>{{ nodes.length }} nodes and {{ edges.length }} edges</small>
+          </span>
+          <!-- Graph settings menu -->
+          <v-menu
+            v-model="graphSettingsMenu"
+            offset-y
+            :close-on-content-click="false"
+            :close-on-click="true"
+            content-class="menu-with-gap"
+          >
+            <template v-slot:activator="{ on, attrs }">
+              <v-btn icon v-bind="attrs" v-on="on" :disabled="!currentGraph" title="Graph settings">
+                <v-icon> mdi-cog-outline </v-icon>
+              </v-btn>
+            </template>
+
+            <v-card class="pa-4 pt-5" width="600">
+              <h5>Layout type</h5>
+              <v-radio-group row v-model="layoutName">
+                <v-radio
+                  v-for="layout in layouts"
+                  :key="layout"
+                  :label="layout"
+                  :value="layout"
+                  @click="buildGraph(currentGraph)"
+                ></v-radio>
+              </v-radio-group>
+
+              <h5>Edge style</h5>
+              <v-radio-group row v-model="edgeStyle">
+                <v-radio
+                  v-for="edge in edgeStyles"
+                  :key="edge"
+                  :label="edge"
+                  :value="edge"
+                  @click="buildGraph(currentGraph)"
+                ></v-radio>
+              </v-radio-group>
+
+              <h5>Transparency for unselected elements</h5>
+              <v-slider v-model="fadeOpacity" @change="changeOpacity" :max="100" :min="0" thumb-label>
+                <template v-slot:thumb-label="{ value }"> {{ value }}% </template>
+              </v-slider>
+            </v-card>
+          </v-menu>
+          <!-- Save graph dialog -->
```

Comment:
When a graph is saved it will be shown in the left panel only after a reload of the page. This feels unintuitive, since there is no feedback where the saved graph went when clicking the save button.
(if this will be fixed in a future version, please open an issue for tracking, thanks)

Reply:
This is fixed. I moved this to the stgate (store) instead. The left panel will now update when new graph is saved.